### PR TITLE
Autobuild amd64

### PR DESCRIPTION
**Pull request by neildhar**
Add build hook to allow automated build for the main image. Note that this will only build the image for amd64 and will push it to the amd64 tag in the registry.

### CODE REVIEW

File: latest/clone-and-build.test.yml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+
+sut:
+  image: ${IMAGE_NAME}
+  command: sudo su duke
+           cd ~/dev && \
+           git clone https://github.com/DukeRobotics/robosub-ros.git && \
+           source /opt/ros/kinetic/setup.bash && \
+           cd robosub-ros/catkin_ws && \
+           catkin build
```
Comment (davidAmiron) [inline latest/clone-and-build.test.yml]: The online image does not contain the catkin directory, so before `catkin build` should be `catkin init`

Comment (davidAmiron) [reply latest/clone-and-build.test.yml]: By catkin directory I mean the .catkin_tools folder that signifies it is a catkin workspace.

Comment (neildhar) [reply latest/clone-and-build.test.yml]: per the [docs for catkin build](https://catkin-tools.readthedocs.io/en/latest/verbs/catkin_build.html):

> If a workspace is not yet initialized, build can initialize it with the default configuration, but only if it is called from the workspace root.

I understand if you think we should initialise anyway for clarity though

Comment (davidAmiron) [reply latest/clone-and-build.test.yml]: oh ok I did not know that, maybe we can still initialise for clarity

Comment (davidAmiron) [reply latest/clone-and-build.test.yml]: Up to you

**Pull request merged**